### ISSUE TITLE
ArgumentException Type must not be ByRef under dotcover

### ISSUE THREAD

**Issue by hazzik** [hazzik/DelegateDecompiler#172]
@hazzik I'm afraid to say that it is not quite fixed 😞I'm seeing a different error based on the changes in 0.29.0 as follows

```
System.ArgumentException: Type must not be ByRef (Parameter 'type')
  Stack Trace:
      at System.Dynamic.Utils.TypeUtils.ValidateType(Type type, String paramName, Boolean allowByRef, Boolean allowPointer)
   at System.Dynamic.Utils.TypeUtils.ValidateType(Type type, String paramName)
   at System.Linq.Expressions.Expression.Convert(Expression expression, Type type, MethodInfo method)
   at System.Linq.Expressions.Expression.Convert(Expression expression, Type type)
   at DelegateDecompiler.Processor.AdjustType(Expression expression, Type type)
   at DelegateDecompiler.Processor.GetArguments(ProcessorState state, MethodBase m)
   at DelegateDecompiler.Processor.Call(ProcessorState state, MethodInfo m)
   at DelegateDecompiler.Processor.Process()
   at DelegateDecompiler.Processor.Process(VariableInfo[] locals, IList`1 args, Instruction instruction, Type returnType)
   at DelegateDecompiler.MethodBodyDecompiler.DecompileConcrete(MethodInfo method, IList`1 args)
   at DelegateDecompiler.MethodBodyDecompiler.Decompile(MethodInfo method, Type declaringType)

```

_Originally posted by @mbrookson in https://github.com/hazzik/DelegateDecompiler/issues/171#issuecomment-773212124_

**Comment by hazzik** [comment]
@mbrookson can you show the debug trace?

**Comment by mbrookson** [comment]
```
Debug Trace:
IL_0000: ldloca.s JetBrains.Profiler.Core.Instrumentation.DataOnStack (0)
 IL_0002: ldc.i4.1
 IL_0003: stfld UInt32 StatementCount
 IL_0008: ldloca.s JetBrains.Profiler.Core.Instrumentation.DataOnStack (0)
 IL_000a: ldc.i4.s 75
 IL_000c: stfld UInt32 MetadataIndex
 IL_0011: ldloca.s JetBrains.Profiler.Core.Instrumentation.DataOnStack (0)
 IL_0013: ldc.i4 100663895
 IL_0018: stfld UInt32 MethodToken
 IL_001d: ldloca.s JetBrains.Profiler.Core.Instrumentation.DataOnStack (0)
 IL_001f: call Void JetBrains_Profiler_Core_Instrumentation_Begin(JetBrains.Profiler.Core.Instrumentation.DataOnStack ByRef)
```

@hazzik Here's what gets logged by default. Let me know if you require more info. Thanks.

**Comment by mbrookson** [comment]
Happy to test stuff in a separate branch if that helps to diagnose and fix the issue.

**Comment by hazzik** [comment]
It's 0:47 here. I would test it tomorrow my time. I think the problem within `AdjustType`. It should not try to adjust type into ref-type (Type.IsRefType).

**Comment by mbrookson** [comment]
No problem. I will do some more investigation and try some things. Thanks!

**Comment by hazzik** [comment]
Ok I've managed to fix this issue. Can you please test this [branch](https://github.com/hazzik/DelegateDecompiler/tree/gh172)?

**Comment by mbrookson** [comment]
Apologies. I didn't see you'd already implemented a fix. Your solution in this [branch](https://github.com/hazzik/DelegateDecompiler/tree/gh172) also works and is definitely better than mine.

Appreciate you spending time on this. Thank you for resolving it! Please update in Nuget when you get a chance.

**Comment by hazzik** [comment]
Released in 0.29.1

**Comment by mbrookson** [comment]
@hazzik Thank you!
